### PR TITLE
[FIX] OWMergeData removes table meta attributes

### LINKED ISSUE #3289: Merge Data removes table meta attributes
##### Orange version
3.16

##### Expected behavior
Merge data should retain table meta attributes (like taxonomy id) from the primary Data Table.

##### Actual behavior
Merge data throws away table meta attributes. 

##### Steps to reproduce the behavior
Run this workflow: [test.ows.zip](https://github.com/biolab/orange3/files/2449733/test.ows.zip) and compare the two Data Infos.
(Requires scOrange).

### PR DESCRIPTION
**Pull request by AndrejaKovacic**
##### Issue

Fixes #3289

##### Description of changes
Merged data is added to original table instead of creating a new one.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by markotoplak** [comment]
Is this changing the input data? The input data should never be modified (if any other widgets were also using the same data they would break). To support workflows, we should never modify data in place.

**Comment by lanzagar** [comment]
What @markotoplak said. And although that could be fixed by making a copy of the data first and modifying that, I don't think you should just overwrite the domain of a Table, the arrays X, Y, metas, etc. It would be too easy to end up with an inconsistent Table (the shape of X might not agree with the size of the domain, ids, ...).

**Comment by AndrejaKovacic** [comment]
I think the most appropriate solution would be to use the function from_numpy and add an option to add metadata as a parameter. I know @lanzagar disagrees with it, because there are some issues with that function, but maybe we should fix from_numpy, instead of trying to create new table here.

**Comment by janezd** [comment]
About travis errors: lint issues are obvious, while to avoid segfault in documentation, you need to rebase to today's master.

File: Orange/widgets/data/owmergedata.py [hunk 1]
```diff
@@ -369,8 +369,12 @@ def _join_table_by_indices(self, reduced_extra, indices):
         string_cols = [i for i, var in enumerate(domain.metas) if var.is_string]
         metas = self._join_array_by_indices(
             self.data.metas, reduced_extra.metas, indices, string_cols)
-        return Orange.data.Table.from_numpy(domain, X, Y, metas)
-
+        table = Orange.data.Table.from_numpy(domain, X, Y, metas)
+        table.name = getattr(self.data, 'name')
+        table.attributes = getattr(self.data, 'attributes')
+        table.ids = getattr(self.data, 'ids')
+        return table
```
Comment (janezd) [inline Orange/widgets/data/owmergedata.py]: I don't know this code well, but judging by what I see in the `Table` class, I think that you should have `getattr(self.data, 'name', '')` and `getattr(self.data, 'attributes', {})`.

`ids` seems to be always present, so it's just `table.ids = self.data.ids`.

(Perhaps `name` and `attributes` are always present, too, but a lot of code does `getattr` for them.)

File: Orange/widgets/data/owmergedata.py [hunk 2]
```diff
@@ -369,7 +369,11 @@ def _join_table_by_indices(self, reduced_extra, indices):
         string_cols = [i for i, var in enumerate(domain.metas) if var.is_string]
         metas = self._join_array_by_indices(
             self.data.metas, reduced_extra.metas, indices, string_cols)
-        return Orange.data.Table.from_numpy(domain, X, Y, metas)
+        table = Orange.data.Table.from_numpy(domain, X, Y, metas)
+        table.name = getattr(self.data, 'name', '')
+        table.attributes = getattr(self.data, 'attributes', '')
```
Comment (janezd) [inline Orange/widgets/data/owmergedata.py]: Default for `attributes` should be `{}`.

(When fixed, don't make a new commit but amend this one, then `push -f`)

**Comment by janezd** [comment]
Burning some midnight oil?

**Pull request merged**